src/napari/layers/_layer_actions.py

```diff
@@ -56,7 +56,11 @@ def _convert(ll: LayerList, type_: str) -> None:
     for lay in list(ll.selection):
         idx = ll.index(lay)
         if isinstance(lay, Shapes) and type_ == 'labels':
-            data = lay.to_labels()
+            # lay.scale is added to the label world extent to account for the virtual width
+            # of each data index (nominally 1.0 for unscaled images)
+            data = lay.to_labels(
+                labels_shape=lay.world_to_data(ll.extent.world[-1] + lay.scale)
+            )
```

Comment:
```suggestion
            data = lay.to_labels(
                labels_shape=ll._extent_world_augmented[-1])
            )
```
Based on @brisvag suggestion, I think this is the way to take into account that extra half pixel, that lay.scale was handling before.

Reply:
This cannot be right. The extent is the bounding box in world coordinates. If you only take the last value (basically the bottom right corner) you ignore anything that's happening elsewhhere. If you *increase* the size of the image following your example above, you're gonna have issues.

What's really missinng is the translate here, not the scale.

Reply:
In fact, if you drag the layerso that the bottom corner is left of the origin, you'll end up with negative shape :P

Reply:
Gah, i totally brain farted while editing the suggestion.
It should be:
`labels_shape=lay.world_to_data(ll._extent_world_augmented[-1])`

Reply:
I just updated the suggestion.

Reply:
I don't think this changes my comment though? You're still only using the bottom right corner of the world extent, ignoring what's going on with the upper left.

Reply:
Yes, sorry. To handle the other corner you need to do something like
https://github.com/napari/napari/pull/8098#issuecomment-3124751770

As is this is just fixing the very confusing behavior where you you can't paint because the Labels are too small, so it's incremental progress.